locations/jclouds/src/main/java/org/apache/brooklyn/location/jclouds/networking/SharedLocationSecurityGroupCustomizer.java

```diff
@@ -0,0 +1,172 @@
+
+    private String locationName = null;
+
+    private List<String> portRanges = null;
+
+    private String cidr = null;
+
+    /**
+     * We track inbound ports from the template to open them once we've created the new
+     * security groups
+     */
+    private int[] inboundPorts;
+
+    /**
+     * The location name is appended to the name of the shared SG - use if you need distinct shared SGs within the same location
+     *
+     * @param locationName appended to the name of the SG
+     */
+    public void setLocationName(String locationName) {
+        this.locationName = locationName;
+    }
+
+    /**
+     * Extra ports to be opened on the entities in the SG
+     *
+     * @param portRanges
+     */
+    public void setPortRanges(List<String> portRanges) {
+        this.portRanges = portRanges;
+    }
+
+    /**
+     * Set to restict the range that the ports that are to be opened can be accessed from
+     *
+     * @param cidr
+     */
+    public void setCidr(String cidr) {
+        this.cidr = cidr;
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, TemplateOptions templateOptions) {
+        super.customize(location, computeService, templateOptions);
+        inboundPorts = templateOptions.getInboundPorts();
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, Template template) {
+        super.customize(location, computeService, template);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (cidr != null) instance.setSshCidrSupplier(Suppliers.ofInstance(new Cidr(cidr)));
+
+        instance.customize(location, computeService, template);
+    }
+
+    @Override
+    public void customize(JcloudsLocation location, ComputeService computeService, JcloudsMachineLocation machine) {
+        super.customize(location, computeService, machine);
+
+        final JcloudsLocationSecurityGroupCustomizer instance = getInstance(getSharedGroupId(location));
+
+        if (portRanges != null) {
+            RangeSet<Integer> portRanges = BrooklynNetworkUtils.portRulesToRanges(this.portRanges);
+
+            List<IpPermission> ipPermissions =
+                    FluentIterable
+                            .from(portRanges.asRanges())
+                            .transform(new Function<Range<Integer>, IpPermission>() {
+                                @Nullable
+                                @Override
+                                public IpPermission apply(@Nullable Range<Integer> integerRange) {
+                                    IpPermission extraPermission = IpPermission.builder()
+                                            .fromPort(integerRange.lowerEndpoint())
+                                            .toPort(integerRange.upperEndpoint())
+                                            .ipProtocol(IpProtocol.TCP)
```

Comment:
Can we support both tcp and upd ports? I presume it would be fairly easy (e.g. change the `portRanges` field and setter to be `tcpPortRanges`, and add a similar field+setter for `udpPortRanges`)?